src/Forms/NumericField.php

```diff
@@ -149,28 +178,36 @@ public function Value()
         return $formatter->format($this->value, $this->getNumberType());
     }
 
-    public function setValue($value, $data = null)
+    public function getValueForValidation(): mixed
     {
-        $this->originalValue = $value;
-        $this->value = $this->cast($value);
-        return $this;
+        $value = $this->value;
+        if (is_numeric($value) && is_string($value)) {
+            return $this->cast($value);
+        }
```

Comment:
Why only call `cast()` if that condition is true?

Reply:
Have removed, we no longer need to cast since the FieldValidator now accepts numeric strings